agent/watch_handler.go

```diff
@@ -21,12 +23,24 @@ const (
 )
 
 // makeWatchHandler returns a handler for the given watch
-func makeWatchHandler(logOutput io.Writer, params interface{}) watch.HandlerFunc {
-	script := params.(string)
+func makeWatchHandler(logOutput io.Writer, handler interface{}) watch.HandlerFunc {
+	script, isScript := handler.(string)
+	args, isArgs := handler.([]string)
+	if isArgs {
+		script = fmt.Sprintf("%v", args)
```

Comment:
This is nasty since we don't use this (and we super don't want to treat the args as a big string. Can you maybe always run the below code based on an args array, so there's less switching around, like this:

```
var args []string
switch h := hander.(type) {
case string:
    // figure out the shell and stuff from the environment
    args = []string{theShell, "-c", h...}
case []string:
    args = h
}

// then down below you just always use ExecSubprocess with the args local
```